cpp/oneapi/dal/algo/basic_statistics/backend/gpu/finalize_compute_kernel_dense_impl_dpc.cpp

```diff
@@ -174,23 +176,44 @@ result_t finalize_compute_kernel_dense_impl<Float>::operator()(const descriptor_
     }
 
     if (res_op_partial.test(result_options::sum)) {
-        const auto sums_nd =
+        auto sums_nd =
             pr::table2ndarray_1d<Float>(q, input.get_partial_sum(), sycl::usm::alloc::device);
+        if (comm_.get_rank_count() > 1) {
```

Comment:
Its better to call it only once, and copy all partial results in one if condition. Also you can potentially get minor perf improvements on 1 gpu, if you call allreduce inside the if.